test/xpu_api/utilities/meta/meta.unary/meta.unary.prop/is_nothrow_assignable.pass.cpp

```diff
@@ -0,0 +1,90 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <class T, class U>
+void
+test_is_nothrow_assignable()
+{
+    static_assert(dpl::is_nothrow_assignable<T, U>::value);
+    static_assert(dpl::is_nothrow_assignable_v<T, U>);
+}
+
+template <class T, class U>
+void
+test_is_not_nothrow_assignable()
+{
+    static_assert(!dpl::is_nothrow_assignable<T, U>::value);
+    static_assert(!dpl::is_nothrow_assignable_v<T, U>);
+}
+
+struct A
+{
+};
+
+struct B
+{
+    void operator=(A);
+};
+
+struct C
+{
+    void
+    operator=(C&); // not const
+};
+
+bool
+kernel_test()
+{
+    test_is_nothrow_assignable<int&, int&>();
+    test_is_nothrow_assignable<int&, int>();
+    test_is_nothrow_assignable<int&, float>();
```

Comment:
I guess we need to add test that for the structure with assignment operator marked as `noexcept` the trait returns `true`.

Reply:
Fixed